packages/@uppy/core/src/Restricter.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable max-classes-per-file, class-methods-use-this */
+/* global AggregateError */
+const prettierBytes = require('@transloadit/prettier-bytes')
+const match = require('mime-match')
+
+const defaultOptions = {
+  maxFileSize: null,
+  minFileSize: null,
+  maxTotalFileSize: null,
+  maxNumberOfFiles: null,
+  minNumberOfFiles: null,
+  allowedFileTypes: null,
+  requiredMetaFields: [],
+}
+
+class RestrictionError extends Error {
+  constructor (...args) {
+    super(...args)
+    this.isRestriction = true
+  }
+}
+
+if (typeof AggregateError === 'undefined') {
+  // eslint-disable-next-line no-global-assign
+  globalThis.AggregateError = class AggregateError extends Error {
```

Comment:
would it be cleaner to instead export this error and let core assign it to the global object? Actually assigning it to the global object is kinda dirty but I guess it's part of our public API so it would be a breaking change if we instead export it from core?

Reply:
Should we instead recommend that users polyfill this class? Or use the core.js polyfill instead of reimplementing it ourselves?

Reply:
What would be the upside? There's probably way more overhead to add a core.js polyfill rather than this 4 liner one.

Reply:
I can see that the implementation in core.js seems more complex than ours: https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/es.aggregate-error.js

maybe our simple implementation is not spec-complete? Then our code could behave differently depending on whether the browser supports it or not

Reply:
OK agreed. Let's add a TODO to remove this in the next major, folks who want to support browsers that have not implemented it can add the polyfill themselves, to align with our policy on polyfills in Uppy.

Reply:
```suggestion
  // TODO: remove this "polyfill" in the next major.
  globalThis.AggregateError = class AggregateError extends Error {
```